### PR TITLE
Add a boundary check for the compiler context menu handler

### LINKED ISSUE #4415: Error: Cannot read properties of undefined (reading 'source')
Sentry Issue: [COMPILER-EXPLORER-A7A](https://sentry.io/organizations/compiler-explorer/issues/3736826413/?referrer=github_integration)

```
Error: Cannot read properties of undefined (reading 'source')

TypeError: Cannot read properties of undefined (reading 'source')
    at t._onContextMenu (https://static.ce-cdn.net/main.v14.2ae5f07e084575279a73.js:1:288220)
    at https://static.ce-cdn.net/v...
  at contextMenuContrib._onContextMenu (./static/panes/compiler.ts:1045:88)
  at Array.forEach (<anonymous>)
...
(12 additional frame(s) were not displayed)
```

### PR DESCRIPTION
**Pull request by jeremy-rifkin**
Fixes #4415

### CODE REVIEW

**Review by RubenRBS — APPROVED** [review]
Great catch!

File: static/panes/compiler.ts [hunk 1]
```diff
@@ -1057,9 +1057,14 @@ export class Compiler extends MonacoPane<monaco.editor.IStandaloneCodeEditor, Co
                         }
                     }
 
-                    const lineSource = this.assembly[e.target.position.lineNumber - 1].source;
+                    if (
+                        e.target.position.lineNumber - 1 >= 0 &&
```
Comment (mattgodbolt) [inline static/panes/compiler.ts:1061]: Huge nitpick but maybe:
```ts
const prevLineNum = e.target.position.lineNumber - 1;
if (prevLineNumber >= 0 && prevLineNumber < this.assembly.length) {
```
etc.

Comment (matsjla) [reply static/panes/compiler.ts:1061]: Sounds reasonable to me too

Comment (jeremy-rifkin) [reply static/panes/compiler.ts:1061]: Good idea thanks

**Pull request merged**